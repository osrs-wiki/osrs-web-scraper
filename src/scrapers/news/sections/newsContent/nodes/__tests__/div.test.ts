import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import divParser from "../div";

describe("div node", () => {
  test("poll-box class should parse and render", () => {
    const root = parse(
      "<div class='poll-box'><p>Question #X:</p><p><b>If While Guthix Sleeps is added to the game, should we continue to explore the idea of offering the Rites of Balance as a rewards from the quest? This question is for developer consideration and not binding.</b></p></div>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([divParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
