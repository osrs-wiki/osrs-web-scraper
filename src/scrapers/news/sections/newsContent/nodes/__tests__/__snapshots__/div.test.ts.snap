// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div node poll-box class should parse and render 1`] = `
"{{News Poll|1|If While Guthix Sleeps is added to the game, should we continue to explore the idea of offering the Rites of Balance as a rewards from the quest? This question is for developer consideration and not binding.}}
"
`;
